tests/test-rest-plugin.php

```diff
@@ -250,4 +250,22 @@ public function test_add_extra_api_taxonomy_arguments() {
 		$this->assertEquals( 'tag', $taxonomy->rest_base );
 		$this->assertEquals( 'WP_REST_Terms_Controller', $taxonomy->rest_controller_class );
 	}
+
+	/**
+	 * The get_rest_url function should return a URL consistently terminated with a "/",
+	 * whether the blog is configured with pretty permalink support or not.
+	 */
+	public function test_rest_url_generation() {
+		$original_structure = get_option( 'permalink_structure' );
+
+		// In pretty permalinks case, we expect a path of wp-json/ with no query
+		update_option( 'permalink_structure', '/%year%/%monthnum%/%day%/%postname%/' );
+		$this->assertEquals( "http://example.org/wp-json/", get_rest_url() );
+
+		// In non-pretty case, we get a query string to invoke the rest router
+		update_option( 'permalink_structure', '' );
+		$this->assertEquals( "http://example.org/?rest_route=/", get_rest_url() );
+
+		update_option( 'permalink_structure', $original_structure );
```

Comment:
I think the option should be restored by PHPUnit's built-in setup/teardown functions, so shouldn't need this.